src/Simplex/Messaging/Agent/Store/SQLite/Util.hs

```diff
@@ -1,382 +1,403 @@
+      ( conn_alias, rcv_host, rcv_port, rcv_id, snd_host, snd_port, snd_id)
+    VALUES
+      (:conn_alias,:rcv_host,:rcv_port,:rcv_id,     NULL,     NULL,   NULL);
+  |]
 
-instance FromField QueueStatus where fromField = fromFieldToReadable
+insertSndQueue_ :: DB.Connection -> SendQueue -> IO ()
+insertSndQueue_ dbConn SendQueue {..} = do
+  let port_ = serializePort_ $ port server
+  DB.executeNamed
+    dbConn
+    insertSndQueueQuery_
+    [":host" := host server, ":port" := port_, ":snd_id" := sndId, ":conn_alias" := connAlias, ":snd_private_key" := sndPrivateKey, ":encrypt_key" := encryptKey, ":sign_key" := signKey, ":status" := status]
 
-instance ToRow ReceiveQueue where
-  toRow ReceiveQueue {rcvId, rcvPrivateKey, sndId, sndKey, decryptKey, verifyKey, status, ackMode} =
-    toRow (rcvId, rcvPrivateKey, sndId, sndKey, decryptKey, verifyKey, status, ackMode)
+insertSndQueueQuery_ :: Query
+insertSndQueueQuery_ =
+  [sql|
+    INSERT INTO snd_queues
+      ( host, port, snd_id, conn_alias, snd_private_key, encrypt_key, sign_key, status)
+    VALUES
+      (:host,:port,:snd_id,:conn_alias,:snd_private_key,:encrypt_key,:sign_key,:status);
+  |]
 
-instance FromRow ReceiveQueue where
-  fromRow = ReceiveQueue undefined <$> field <*> field <*> field <*> field <*> field <*> field <*> field <*> field
+insertSndConnection_ :: DB.Connection -> SendQueue -> IO ()
+insertSndConnection_ dbConn SendQueue {server, sndId, connAlias} = do
+  let port_ = serializePort_ $ port server
+  DB.executeNamed
+    dbConn
+    insertSndConnectionQuery_
+    [":conn_alias" := connAlias, ":snd_host" := host server, ":snd_port" := port_, ":snd_id" := sndId]
 
--- TODO refactor into a single query with join
-getRcvQueue :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> QueueRowId -> m ReceiveQueue
-getRcvQueue st@SQLiteStore {conn} queueRowId = do
-  r <-
-    liftIO $
-      DB.queryNamed
-        conn
-        [sql|
-        SELECT server_id, rcv_id, rcv_private_key, snd_id, snd_key, decrypt_key, verify_key, status, ack_mode
-          FROM receive_queues
-          WHERE receive_queue_id = :rowId;
-        |]
-        [":rowId" := queueRowId]
-  case r of
-    [Only serverId :. rcvQueue] ->
-      (\srv -> (rcvQueue {server = srv} :: ReceiveQueue)) <$> getServer st serverId
-    _ -> throwError SENotFound
+insertSndConnectionQuery_ :: Query
+insertSndConnectionQuery_ =
+  [sql|
+    INSERT INTO connections
+      ( conn_alias, rcv_host, rcv_port, rcv_id, snd_host, snd_port, snd_id)
+    VALUES
+      (:conn_alias,     NULL,     NULL,   NULL,:snd_host,:snd_port,:snd_id);
+  |]
 
-getRcvQueueByRecipientId :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> RecipientId -> HostName -> Maybe ServiceName -> m ReceiveQueue
-getRcvQueueByRecipientId st@SQLiteStore {conn} rcvId host port = do
+retrieveConnQueues :: DB.Connection -> ConnAlias -> IO (Maybe ReceiveQueue, Maybe SendQueue)
+retrieveConnQueues dbConn connAlias =
```

Comment:
maybe make this one without transaction and use withTransaction where needed?

Reply:
`retrieveConnQueues` is exported, the point in adding `retrieveConnQueuesTransactionless_` was to keep the former transactional, and to have the latter reused inside other utility functions. The only point where it could be made transactional elsewhere is on SQLite.hs above but then it kind of ruins splitting logic and physical implementation between it and Util.hs

Reply:
renamed to `retrieveConnQueues_`